src/main/java/net/sf/jabref/MetaData.java

```diff
@@ -83,26 +83,17 @@
     private MetaData(Map<String, String> inData, Charset encoding) throws ParseException {
         Objects.requireNonNull(inData);
         setData(inData);
-        this.encoding = encoding;
+        this.encoding = Objects.requireNonNull(encoding);
     }
 
     /**
-     * The MetaData object can be constructed with no data in it.
+     * The MetaData object can be constructed with no data in it, but needs encoding information
```

Comment:
Why is the encoding special and always has to be provided while other metadata is optional? I would still provide an empty constructor and let `getEncoding` return an `Optional<Encoding>`. Then it is the caller business to decide which encoding he wants if no encoding is set in the metadata.